velox/common/caching/CachedFactory.h

```diff
@@ -348,59 +160,32 @@ CachedFactory<Key, Value, Generator, Sizer, Comparator, Hash>::generate(
       pendingCv_.notify_all();
       throw;
     }
-    const uint64_t sizeOccupied = Sizer()(*generatedValue);
-    Value* rawValue = generatedValue.release();
     cacheMu_.lock();
-    const bool inserted = cache_->addPinned(key, rawValue, sizeOccupied);
+    cache_->add(key, generatedValue);
```

Comment:
This can fix an issue that happened before:
If cache is full of pinned entries, the cache will refuse to evict. New entry added will not be noted by the cache. This can create a scenario in which a large number of outstanding cache entries not aware by the cache (and might cause memory issues).